workitems/l2-transaction-data-fee/l2-transaction-data-fee-v1.0-psd01.md

```diff
@@ -196,55 +264,120 @@ Note that requirements are uniquely numbered in ascending order within each requ
+###### L2TFSPECCONCEPT
+
+To specify requirements about a L2 Transaction Data Fee, this document will first:
+
+* Define the meaning of an L2 Transaction Data Fee
+* Define requirements requirements on an L2 Transaction Data Fee and its relationships to L2 transaction fee types.
+
+Note, that all definitions can be found in the [Glossary](#12-glossary)
+
+The specification section will define requirements around Transaction Data Fee transparency, estimation, final amount, and how, when and where such a fee is communicated to different roles.
+
+## 2.1 Definition of an L2 Transaction Data Fee
+###### L2TFSPECDEF
+
+This document defines an L2 Transaction Data Fee as the portion of the Execution fee in a Layer 2 (L2) network or protocol token to be paid by a Transaction Originator or Transaction Sender that is derived from the amount of L2 transaction data stored on the L1.
+
+This documents follows the definitions of Transaction, Account, State and State Transition in [L2 Transaction Fee Specification](https://github.com/eea-oasis/L2/blob/main/workitems/l2-transaction-fees/l2-transaction-fees-v1.0-psd01.md). [Note: This is not yet the merged version of PR 43]
+
+Note that a Transaction Data Fee has an L1 Gas Price which is defined in the context of this document as an L2 Gas price or a price of an L2 gas equivalent unit of compute and storage consumption based on L2 transaction data stored on the L1.
+
+Note that a Gas price or a price of an L2 gas equivalent unit of compute and storage consumption is typically variable and changes with the level of usage of an L2/L1 network. Note that a gas price is expressed in Giga Wei (GWEI) for EVM-compatible networks, where Wei represents the smallest unit of gas; 1 Wei = 10<sup>-18</sup> Eth in Ethereum's native token.
+
+Furthermore, note that Gas in this document refers to the unit that measures the amount of computational and storage effort required to execute specific operations on an EVM-compatible network.
+
+## 2.2 Definition of L2 Roles relevant to an L2 Transaction Data Fee
+###### L2TFSPECDEFROLES
+
+This document follows the role definitions in the [L2 Transaction Fee Specification](https://github.com/eea-oasis/L2/blob/main/workitems/l2-transaction-fees/l2-transaction-fees-v1.0-psd01.md). [Note: This is not yet the merged version of PR 43]
+
+## 2.3 Definition of Transaction Types
+###### L2TFSPECDEFTTYPES
 
-TBD
+This document follows the definitions of transaction types in the [L2 Transaction Fee Specification](https://github.com/eea-oasis/L2/blob/main/workitems/l2-transaction-fees/l2-transaction-fees-v1.0-psd01.md). [Note: This is not yet the merged version of PR 43]
 
 -------
 
-# 3 Layer 2 Transaction Fees Specification
+# 3 Layer 2 Transaction Data Fee Specification
+###### L2TFSPECSEC
 
-TBD
+#### **[R1]**
+An L2 Transaction Data Fee MUST be indicated to be part of the Execution Fee.
+
+#### **[R2]**
+An L2 Transaction Data Fee MUST be displayed separately from the Execution Fee.
```

Comment:
Conflicting with R1?

Reply:
@smartcontracts not really ... you can list the execution fee something like this `(EF: 10, DF: 6, Rest: 4)` that meets R1 and R2